src/core/agents.jl

```diff
@@ -57,60 +71,168 @@ mutable struct Person{T} <: AbstractAgent
 end
+```
+### Exaple with optional hierachy
+An alternative way to make the above structs, that also establishes
+a user-specific subtyping hierachy would be to do:
+```julia
+abstract type AbstractHuman <: AbstractAgent end
 
-"""
-    GridAgent{D}
-Combine with [`@agent`](@ref) to create an agent type for `D`-dimensional
-[`GridSpace`](@ref). It attributes the fields `id::Int, pos::NTuple{D,Int}`
-to the start of the agent type.
-"""
-mutable struct GridAgent{D} <: AbstractAgent
-    id::Int
-    pos::Dims{D}
+@agent Worker GridAgent{2} AbstractHuman begin
+    age::Int
+    moneyz::Float64
 end
 
+@agent Fisher Worker AbstractHuman begin
+    fish_per_day::Float64
+end
+```
+which would now make both `Human, Fisher` subtypes of `AbstractHuman`.
+
+### Example highlighting problems with parametric types
+Notice that in Julia parametric types are union types.
+Hence, the following cannot be used:
+```julia
+@agent Dummy{T} GridAgent{2} begin
+    moneyz::T
+end
+
+@agent Fisherino{T} Dummy{T} begin
+    fish_per_day::T
+end
+```
+You will get an error in the definition of `Fisherino`, because the fields of
+`Dummy{T}` cannot be obtained, because it is a union type. Same with using `Dummy`.
+You can only use `Dummy{Float64}`.
+
+### Example with common dispatch and no subtyping
+It may be that you do not even need to create a subtyping relation if you want
+to utilize multiple dispatch. Consider the example:
+```julia
+@agent CommonTraits GridSpace{2} begin
+    age::Int
+    speed::Int
+    energy::Int
+end
+```
+and then two more structs are made from these traits:
+```julia
+@agent Bird CommonTraits begin
+    height::Float64
+end
+
+@agent Rabbit CommonTraits begin
+    underground::Bool
+end
+```
+
+If you wanted a function that dispatches to both `Rabbit, Bird`, you only have to define:
+```julia
+Animal = Union{Bird, Rabbit}
+f(x::Animal) = ... # uses `CommonTraits` fields
+```
+However, it should also be said, that there is no real reason here to explicitly
+type-annotate `x::Animal` in `f`. Don't annotate any type. Annotating a type
+only becomes useful if there are at least two "abstract" groups, like `Animal, Person`.
+Then it would make sense to define
+```julia
+Person = Union{Fisher, Baker}
+f(x::Animal) = ... # uses `CommonTraits` fields
+f(x::Person) = ... # uses fields that all "persons" have
+```
 """
-    ContinuousAgent{D}
-Combine with [`@agent`](@ref) to create an agent type for `D`-dimensional
-[`ContinuousSpace`](@ref). It attributes the fields
-`id::Int, pos::NTuple{D,Float64}, vel::NTuple{D,Float64}`
-to the start of the agent type.
-"""
-mutable struct ContinuousAgent{D} <: AbstractAgent
-    id::Int
-    pos::NTuple{D,Float64}
-    vel::NTuple{D,Float64}
+macro agent(new_name, base_type, extra_fields)
+    # This macro was generated with the guidance of @rdeits on Discourse:
+    # https://discourse.julialang.org/t/
+    # metaprogramming-obtain-actual-type-from-symbol-for-field-inheritance/84912
```

Comment:
```suggestion
    # https://discourse.julialang.org/t/metaprogramming-obtain-actual-type-from-symbol-for-field-inheritance/84912
```

So it's easier to copy-paste the link